src/actions.ml

```diff
@@ -2075,3 +2086,87 @@ let push_action ~bot_info ~base_ref ~commits_msg =
     else Lwt.return ()
   in
   Lwt_list.iter_s commit_action commits_msg
+
+let days_elapsed ts =
+  (* Yes, I know this is wrong because of DST and black holes but it should
+     still be correct enough *)
+  Float.to_int ((Unix.time () -. ts) /. (3600. *. 24.))
+
+let rec apply_throttle len action args =
+  if List.is_empty args || len <= 0 then Lwt.return ()
+  else
+    let args, rem = List.split_n args len in
+    Lwt_list.map_p action args >>= fun ans ->
+    let n = List.count ~f:(fun b -> b) ans in
+    apply_throttle (len - n) action rem
+
+let apply_after_label ~bot_info ~owner ~repo ~after ~label ~action ?throttle () =
+  GitHub_queries.get_open_pull_requests_with_label ~bot_info ~owner ~repo ~label
+  >>= function
+  | Ok prs ->
+    let iter (pr_id, pr_number) =
+      GitHub_queries.get_pull_request_label_timeline ~bot_info ~owner ~repo ~pr_number >>= function
+      | Ok timeline ->
```

Comment:
This isn't a change request, but we could optimize the number of API calls a lot by also returning say the last 10 timeline labeling operations within the `get_open_pull_requests_with_label` GraphQL query. Then, we would need to request the full label timeline only for the PRs for which the label was not set among the last 10 operations.